contracts/Oracle.sol

```diff
@@ -68,183 +68,165 @@ contract Oracle is Ownable {
+    function isLockingPeriodOver(address _asset, uint256 _expiryTimestamp) public view returns (bool) {
+        address pricer = assetPricer[_asset];
+        uint256 lockingPeriod = pricerLockingPeriod[pricer];
 
         return now > _expiryTimestamp.add(lockingPeriod);
     }
 
     /**
      * @notice check if dispute period is over
-     * @param _batch batch hash
-     * @param _expiryTimestamp batch expiry
+     * @param _asset assets to query
+     * @param _expiryTimestamp expiry timestamp of otoken
      * @return True if dispute period is over, otherwise false
      */
-    function isDisputePeriodOver(bytes32 _batch, uint256 _expiryTimestamp) public view returns (bool) {
-        address oracle = batchOracle[_batch];
-        uint256 disputePeriod = oracleDisputePeriod[oracle];
-
-        Price memory batchPrice = batchPriceAt[_batch][_expiryTimestamp];
-
-        if (batchPrice.timestamp == 0) {
+    function isDisputePeriodOver(address _asset, uint256 _expiryTimestamp) public view returns (bool) {
+        // check if the pricer has submitted the price at this tiestamp
+        Price memory price = storedPrice[_asset][_expiryTimestamp];
+        if (price.timestamp == 0) {
             return false;
         }
 
-        return now > batchPrice.timestamp.add(disputePeriod);
+        address pricer = assetPricer[_asset];
+        uint256 disputePeriod = pricerDisputePeriod[pricer];
+        return now > price.timestamp.add(disputePeriod);
     }
 
     /**
-     * @notice set batch oracle
+     * @notice set pricer for an asset
      * @dev can only be called by owner
-     * @param _batch batch (hash of underlying, stike, collateral and expiry)
-     * @param _oracle oracle address
+     * @param _asset asset
+     * @param _pricer pricer address
      */
-    function setBatchOracle(bytes32 _batch, address _oracle) external onlyOwner {
-        batchOracle[_batch] = _oracle;
+    function setAssetPricer(address _asset, address _pricer) external onlyOwner {
+        require(_pricer != address(0), "Oracle: cannot set pricer to address(0)");
+        assetPricer[_asset] = _pricer;
 
-        emit BatchOracleUpdated(_batch, _oracle);
+        emit PricerUpdated(_asset, _asset);
     }
 
     /**
-     * @notice set oracle locking period
-     * @dev can only be called by owner
-     * @param _oracle oracle address
+     * @notice set pricer locking period
+     * @dev this function can only be called by owner
+     * @param _pricer pricer address
      * @param _lockingPeriod locking period
      */
-    function setLockingPeriod(address _oracle, uint256 _lockingPeriod) external onlyOwner {
-        oracleLockingPeriod[_oracle] = _lockingPeriod;
+    function setLockingPeriod(address _pricer, uint256 _lockingPeriod) external onlyOwner {
+        pricerLockingPeriod[_pricer] = _lockingPeriod;
 
-        emit OracleLockingPeriodUpdated(_oracle, _lockingPeriod);
+        emit PricerLockingPeriodUpdated(_pricer, _lockingPeriod);
```

Comment:
should probably not allow locking period to be reset during locking period. Admin doesn't need that much power do they? Also it seems like a malicious admin could then keep locking up people's funds.

Reply:
Same with dispute period, should not allow reset of the dispute period during a dispute period.

Reply:
or should have some max caps on these?

Reply:
Do not allow setting dispute period during dispute period sounds good to me.